### ISSUE TITLE
VS 2015 Update 3 (with patch for update 3 installed) crashes

### ISSUE THREAD

**Issue by stefan505** [Microsoft/TypeScript#9727]
**TypeScript Version**: 1.8.10

**VS 2015**
Update 3 (with patch for Update 3)

**Expected behavior:**
TypeScript debugging should step into JS source and be able to view stack traces during debugging.

**Actual behavior:** 
VS 2015 crashes as soon as I step into JS source and view stack traces during debugging. Discovered it also crashes when adding breakpoints and encountering an unhandled JS error.

```
Application: devenv.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.NullReferenceException
   at TypeScriptDebugEngine.SymbolProvider.DecodedSourceMapDataItem.MapJsSourcePosition(Microsoft.VisualStudio.Debugger.Symbols.DkmTextSpan, Microsoft.VisualStudio.Debugger.Script.DkmScriptDocument)
   at TypeScriptDebugEngine.SymbolProvider.SourcePositionResolver.Microsoft.VisualStudio.Debugger.ComponentInterfaces.IDkmSymbolQuery.GetSourcePosition(Microsoft.VisualStudio.Debugger.Symbols.DkmInstructionSymbol, Microsoft.VisualStudio.Debugger.Symbols.DkmSourcePositionFlags, Microsoft.VisualStudio.Debugger.Evaluation.DkmInspectionSession, Boolean ByRef)
   at Microsoft.VisualStudio.Debugger.EntryPoint.IDkmSymbolQuery_GetSourcePosition(IntPtr, IntPtr, Microsoft.VisualStudio.Debugger.Symbols.DkmSourcePositionFlags, IntPtr, Byte ByRef, IntPtr ByRef)

```

I have also installed [this update](https://msdn.microsoft.com/en-us/library/mt752379.aspx?f=255&MSPPError=-2147217396) for VS 2015 Update 3 with no effect.

**Comment by stefan505** [comment]
I see its been logged before [here](https://github.com/Microsoft/TypeScript/issues/9524), but I assume it has been fixed, but I'm still getting the error.

**Comment by stefan505** [comment]
Investigated it further, it happens as soon as I try to step into JS source code that is outside my TS debugging scope. Issue confirmed on secondary workstation with exactly the same setup as mine. Workstations on Update 2 is not encountering this issue.

**Comment by sheetalkamat** [comment]
Looking into it

**Comment by mhegazy** [comment]
We believe the issue should be fixed in the latest update3 released, as well as in TS 2.0 RC and later. please reopen if this is not the case.
